sympy/ntheory/tests/test_residue.py

```diff
@@ -127,7 +127,11 @@ def test_residue():
     #issue 10816
     assert is_nthpow_residue(1, 0, 1) is False
     assert is_nthpow_residue(1, 0, 2) is True
-    assert is_nthpow_residue(2, 0, 2) is False
+    assert is_nthpow_residue(3, 0, 2) is False
+    assert is_nthpow_residue(0, 1, 8) is True
+    assert is_nthpow_residue(2, 3, 2) is False
+    assert is_nthpow_residue(2, 3, 9) is False
+    assert is_nthpow_residue(9, 6, 16) is False
```

Comment:
No, just a mistake in what I was testing. I was looking for an example where `return pow(a, k, m) == 1` is False but I had kept `==` instead of using `!=`. The previous test for 2, 3, 9 satisfies that condition so the 9,6,16 line should be removed. Other than that I have no further tests to suggest. I'll cancel tests, open a PR with the corrected line and commit from there once tests finish.